clar2wasm/src/words/conditionals.rs

```diff
@@ -369,6 +370,75 @@ impl Word for Or {
     }
 }
 
+#[derive(Debug)]
+pub struct Unwrap;
+
+impl Word for Unwrap {
+    fn name(&self) -> ClarityName {
+        "unwrap!".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let input = args.get_expr(0)?;
+        let throw = args.get_expr(1)?;
+
+        generator.traverse_expr(builder, input)?;
+
+        let throw_type = clar2wasm_ty(generator.get_expr_type(throw).expect("Throw must be typed"));
+
+        let inner_type = match generator.get_expr_type(input) {
+            Some(TypeSignature::OptionalType(inner_type)) => (**inner_type).clone(),
+            Some(TypeSignature::ResponseType(inner_types)) => {
+                let (ok_type, err_type) = &**inner_types;
+                // Drop the err value;
+                drop_value(builder, &err_type);
+                ok_type.clone()
+            }
+            _ => return Err(GeneratorError::TypeError("Invalid type for unwrap".into())),
+        };
+
+        // stack [ discriminant some_val ]
+        let some_locals = generator.save_to_locals(builder, &inner_type, true);
+
+        let mut throw_branch = builder.dangling_instr_seq(InstrSeqType::new(
+            &mut generator.module.types,
+            &[],
+            &throw_type,
+        ));
+        generator.traverse_expr(&mut throw_branch, throw)?;
+        throw_branch.return_();
```

Comment:
I think if we return here, we'll miss restoring the stack pointer. Instead, we just want to exit the block.

See https://github.com/stacks-network/clarity-wasm/blob/a235fc37320f2d3eda6f08251f2bef34c6347c36/clar2wasm/src/words/functions.rs#L124-L135

Reply:
There may be some missing infrastructure for this that isn't there yet (hence the TODO). Like, I think we may want to store that block id in the generator or something.